fastify.js

```diff
@@ -298,16 +286,27 @@ function build (options) {
     }
 
     req.id = genReqId(req)
-    handleTrustProxy(req)
-    req.hostname = req.hostname || req.headers['host']
+
+    var hostname = req.headers['host']
+    var ip = req.connection.remoteAddress
+    var ips
+
+    if (trustProxy) {
+      ip = proxyAddr(req, proxyFn)
+      ips = proxyAddr.all(req, proxyFn)
+      if (ip !== undefined) {
+        hostname = req.headers['x-forwarded-host']
```

Comment:
If the `x-forwarded-host` header is not available, the `host` header should be used as a fallback.
A test should be added for this case.

Reply:
Are you referring to something like ```
 if (ip !== undefined && req.headers['x-forwarded-host']) {
        hostname = req.headers['x-forwarded-host']
``` ?

And do you know what does this mean? Not sure how to do this...
> Can you please make this change non semver major? The properties must still be on the raw objects.

Reply:
> Are you referring to something like
>
> ```
> if (ip !== undefined && req.headers['x-forwarded-host']) {
>        hostname = req.headers['x-forwarded-host']
> ```
> ?

Yes, that looks good 👍

> And do you know what does this mean?

To "make this change non semver major" means to make changes to this PR so that it doesn't break any existing code (so the PR can be accepted *without* needing to increase the [semver](https://semver.org/) major number).

To do this, you just need to add the `ip`, `ips`, and `hostname` properties back to the Node `req` object. That way, any code that currently relies on those properties won't break after this PR is merged. The properties can be removed in a separate PR (which will get labelled as "semver major").

Reply:
@nwoltman Thanks! A test is added in 3ce1c00. I also added ip, ips, and hostname back in 933d4d3, but the whole reason for this PR is to make fastify functional on Google Cloud Function. https://github.com/fastify/fastify/issues/946#issuecomment-463037727. So is it in your roadmap that the removal can only happen at 3.x.x? And if so, when will that be?

Reply:
Yes, the removal will happen only for v3.0.0. There currently isn't a plan for when v3 will be released, but as the [LTS schedule](https://github.com/fastify/fastify/blob/master/docs/LTS.md) states, v2 will be supported for at least 6 months, so v3 won't be realeased for at least that long.

It's an unfortunate situation, but all we can do is wait. In the meantime, at least you can rely on [this workaround](https://github.com/fastify/fastify/issues/946#issuecomment-469540097).